text/006-class-name.md

```
@@ -0,0 +1,405 @@
+
+buttons.light({ size: "small" });
+// => "rounded border px-2 py-1 text-sm bg-white text-black"
+
+buttons.dark({ size: "large" });
+// => "rounded border px-4 py-2 text-lg bg-gray-900 text-white"
+```
+
+This allows you to create reusable variant factories with type-safe parameters, similar to `rules.with()`.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
```

markdownlint-cli2 (0.18.1)

Link and image reference definitions should be needed
Unused link or image reference definition: "reference-level-explanation"

(MD053, link-image-reference-definitions)